### PR TITLE
BUILD(docker): Updated Dockerfile to use CMake

### PR DESCRIPTION
**Pull request by Avatat**
The change fixes this issue: https://github.com/mumble-voip/mumble/issues/4514

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -27,10 +28,10 @@ RUN apt-get update && apt-get install -y \
 	qtcreator
 
 COPY . /root/mumble
-WORKDIR /root/mumble
+WORKDIR /root/mumble/release
 
-RUN qmake -recursive main.pro CONFIG+="no-client grpc"
-RUN make release
+RUN cmake -Dclient=OFF -Dgrpc=ON ..
```
Comment (Krzmbrzl) [inline Dockerfile:33]: I would explicitly use `-DCMAKE_BUILD_TYPE=Release` here. Then the Docker file does not rely on default values for the build type.

File: Dockerfile [hunk 2]
```diff
@@ -27,10 +28,10 @@ RUN apt-get update && apt-get install -y \
 	qtcreator
 
 COPY . /root/mumble
-WORKDIR /root/mumble
+WORKDIR /root/mumble/release
```
Comment (Krzmbrzl) [inline Dockerfile:31]:
```suggestion
WORKDIR /root/mumble/build
```

`build` is the usual name for the build-dir used for cmake. In the end this doesn't really matter though :shrug:

**Review by Krzmbrzl — CHANGES_REQUESTED** [review]
Also please make sure you follow our [commit guidelines](https://github.com/mumble-voip/mumble/blob/master/COMMIT_GUIDELINES.md).

I would probably put it into the `BUILD(docker)` type & scope. And the issue that is being fixed sould also be referenced in your commit message (as shown in the linked guidelines)

**Comment by Avatat** [comment]
@Krzmbrzl, thank you for your comments!
I'm a little bit outdated, sorry for the mistakes. I will correct it.

File: Dockerfile [hunk 3]
```diff
@@ -27,10 +28,10 @@ RUN apt-get update && apt-get install -y \
...
 
-RUN qmake -recursive main.pro CONFIG+="no-client grpc"
-RUN make release
+RUN cmake -Dclient=OFF -Dgrpc=ON ..
+RUN make -j $(nproc)
```
Comment (davidebeatrici) [inline Dockerfile:34]: I would use Ninja instead.

Comment (Krzmbrzl) [reply Dockerfile:34]: Unnecessary. Ninja isn't faster and is only a dependency that has to be installed in order for the Dockerfile to work. Make on the other hand is part of every Linux distro (that is capable of compiling in the first place)

Comment (ZeroAbility) [reply Dockerfile:34]: I am in favor for whichever provides the least effort in terms of building the image. As far as Ninja being faster or not, it really depends on the size of the project. I have seen faster builds with it when building both client and server. If the difference is negligible when building just the server, then `make` should be fine.

Comment (Krzmbrzl) [reply Dockerfile:34]: For a fresh build there seems to be almost no difference between make and Ninja: https://david.rothlis.net/ninja-benchmark/
@ZeroAbility are you perhaps referring to Ninja being faster than the default MSBuild on Windows?

Comment (ZeroAbility) [reply Dockerfile:34]: > are you perhaps referring to Ninja being faster than the default MSBuild on Windows?

Probably, NMake. If there isn't a difference then I would also suggest make.

File: Dockerfile (hunk 2 above)
Comment (Avatat) [reply Dockerfile]: I used `release` because it was used previously in Dockerfile. I changed it to `build`.

File: Dockerfile (hunk 1 above)
Comment (Avatat) [reply Dockerfile]: I added `-DCMAKE_BUILD_TYPE=Release` to CMake options.

**Comment by Avatat** [comment]
@Krzmbrzl, I updated my PR, following commit guidelines.

**Review by Krzmbrzl — APPROVED** [review]

**Review by davidebeatrici — APPROVED** [review]

**Pull request merged**